apps/account/lib/account/controller/account.ex

```diff
@@ -23,30 +26,33 @@ defmodule Helix.Account.Controller.Account do
     end
   end
 
-  @spec find_by([email: Account.email]) :: {:ok, Account.t} | {:error, :notfound}
-  def find_by(email: email) do
-    email = String.downcase(email)
+  @spec find_by(find_params) :: [Account.t]
+  def find_by(params) do
+    query = Enum.reduce(params, Account, &reduce_find_params/2)
 
-    case Repo.get_by(Account, email: email) do
-      nil ->
-        {:error, :notfound}
-      account ->
-        {:ok, account}
-    end
+    Repo.all(query)
   end
 
-  @spec update(Account.id, Account.update_params) :: {:ok, Account}
-    | {:error, Ecto.Changeset.t}
-    | {:error, :notfound}
-  def update(account_id, params) do
-    with {:ok, account} <- find(account_id) do
-      account
-      |> Account.update_changeset(params)
-      |> Repo.update()
-    end
+  @spec reduce_find_params({:email, String.t}, Ecto.Queryable.t) ::
```

Comment:
2 bad that this is getting removed [here](https://github.com/kress95/Helix/tree/fix-account-finds)